nat/traversal/pinger.go

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package traversal
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"time"
+
+	log "github.com/cihub/seelog"
+	"github.com/pkg/errors"
+	"golang.org/x/net/ipv4"
+)
+
+const prefix = "[NATPinger] "
+const pingInterval = 200
+
+type ConfigParser interface {
+	Parse(config json.RawMessage) (ip string, port int, err error)
+}
+
+// Pinger represents NAT pinger structure
+type Pinger struct {
+	localPort     int
+	pingTarget    chan json.RawMessage
+	pingReceived  chan bool
+	pingCancelled chan bool
```

Comment:
once again bool channels, don't think we need the true and false values here, a struct{} is enough to indicate that.